### PR TITLE
[air] Move to new storage_path API in tests and examples

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

Following #33463, this PR updates our tests, examples, and docs to use the new `storage_path` API.

The only locations where we continue to use the `local_dir` statement are tests where we specify both a local dir and a remote dir. For these tests, we can move to an environment-variable based wrapper in the future.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by xwjiang2010 — APPROVED** [review]

File: doc/source/tune/tutorials/tune-storage.rst [hunk 1]
```diff
@@ -281,7 +279,7 @@ Again, we're running this example script from the Ray cluster's head node.
         my_trainable,
         run_config=air.RunConfig(
             name="my-tune-exp",
-            local_dir="/tmp/mypath",
+            storage_path="/tmp/mypath",
             # Use the default syncing behavior
             # You don't have to pass an empty sync config - but we
             # do it here for clarity and comparison
```
Comment (justinvyu) [inline doc/source/tune/tutorials/tune-storage.rst:285]: We can remove this empty `SyncConfig`. We don't pass it in anymore in the examples above.

**Review by justinvyu — COMMENTED** [review]
Looks good.

I do have a question about a `SyncConfig.upload_dir` remainder that is causing some unintended effects:

Experiment checkpoint upload logic currently checks for `upload_dir` in order to determine whether or not to ignore uploading checkpoint dirs. We should just check for `self._remote_checkpoint_dir` and remove the need for sync_config in this case: https://github.com/ray-project/ray/blob/a9431c7e4a2846a773cbd9a689bb23db0a0d7cc8/python/ray/tune/execution/experiment_state.py#L237 

Should we fix this in the current PR? Or is there another PR queued up already?

**Comment by krfricke** [comment]
The test failure seems unrelated - it's a segfault that seems to come from a different source. It also seems to be a flaky test. I've left some debug prints in the test for the next time this failure occurs, but assuming the unit tests pass, we can merge this PR.

**Review by richardliaw — APPROVED** [review]
stamp docs

**Pull request merged**